### PR TITLE
Fixes for animate docs

### PR DESCRIPTION
**Pull request by FRSgit**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Updated animaton-aware directive tables both on ngAnimate's module and on animate guide pages.

**What is the current behavior? (You can also link to an open issue here)**
That's an effect of discussion which took place under #16561 

**What is the new behavior (if this is a feature change)?**
No, just docs.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [x] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

### CODE REVIEW

**Comment by FRSgit** [comment]
I signed it!

**Review by gkalpak — CHANGES_REQUESTED** [review]
LGTM except for the (re-)ordering. My guess is that you tried to switch to some sort of alphabetic ordering, but it doesn't seem to have worked out very well 😛

**Comment by FRSgit** [comment]
Yup, I've reordered it, so they appear in the same order as in navigation of the docs, what's bad about it?

**Comment by FRSgit** [comment]
> Previously, they were grouped by animation type (e.g. structural vs non-structural), but not ordered other than that (afaict). Also, similar directives where close to each other (e.g. ngHide/ngShow).
Ah, I get it now. Well, then I'll put them alphabetically since it's the only obvious (for reader) way of handling it I guess (the *structural* naming is not so openly described in docs).
Still, the main focus there was to sync table from naAnimate page & animation guide.

**Comment by FRSgit** [comment]
Tables reordered & unified.

File: src/ngAnimate/module.js [hunk 1]
```diff
@@ -17,20 +17,28 @@
...
+ * | {@link ng.directive:ngModel#animations ngModel}                               | add and remove ({@link ng.directive:ngModel#css-classes various classes}) |
+ * | {@link ng.directive:ngRepeat#animations ngRepeat}                             | enter, leave, and move                                                    |
+ * | {@link ng.directive:ngShow#animations ngShow}                                 | add and remove (the `ng-hide` class)                                      |
+ * | {@link ng.directive:ngSwitch#animations ngSwitch}                             | enter and leave                                                           |
+ * | {@link ngRoute.directive:ngView#animations ngView}                            | enter and leave                                                           |
+ *
+ * (More information can be found by visiting the documentation associated with each directive.)
+ *
+ * For a full breakdown of the steps involved during each animation event, refer to the
+ * {@link ng.$animate API docs}.
```
Comment (gkalpak) [inline src/ngAnimate/module.js]: The text of the link will `API docs` (which is kind of vague). Maybe change it to `` `$animate` API docs``?

**Review by gkalpak — APPROVED** [review]
LGTM (with one minor suggestion) 👍

File: src/ngAnimate/module.js (hunk 1 above)
Comment (FRSgit) [reply src/ngAnimate/module.js]: Sure, just copied & pasted it from animations guide, wait for the commit ⌛️

**Comment by FRSgit** [comment]
Done!

**Pull request closed without merging**